src/magic_folder/uploader.py

```diff
@@ -254,8 +253,10 @@ def upload_local_snapshots(self):
         localsnapshot_names = self._config.get_all_localsnapshot_paths()
 
         # update our status if we have nothing to do
-        if not len(localsnapshot_names):
-            self._status.upload_stopped()
+        if len(localsnapshot_names):
+            self._status.upload_started(self._config.name)
+        else:
+            self._status.upload_stopped(self._config.name)
 
         # XXX: processing this table should be atomic. i.e. While the upload is
         # in progress, a new snapshot can be created on a file we already uploaded
```

Comment:
We may eventually want to check again at the end of the loop (otherwise we'll need to wait for poll_interval for syncing to be done. I think we should leave a comment (and maybe issue) but leave the complexity of doing it twice for now.